### PR TITLE
V0.4dev1.1

### PR DESCRIPTION
**Pull request by kurtkeller**
a number of suggested improvements

- I bumped the version to a development level, as I have some more
  suggested adaptions in the pipeline and in my head. But I am not
  sure how you want your versioning to be done.

  This is a separate commit, you you can easily skip it, if you are not
  happy with this versioning.

  Possibly you might want to create a development branch (one of
  the changes in my pipeline depends on changes in another library,
  all of them but one already accepted and commited, but not yet
  released).

- Added help texts to the argument parser.
  
  Also restricted some arguments to some predefined choices
  where only those choices are valid.

- Restrict accepted account (IBAN) numbers to CH and LI; according
  to the online specifications, these are the only ones accepteable.

- Fix a runtime error when no --extra-info is given.
  len(extra_infos) is invalid when None is passed to __init__ from the
  argument parser for extra_infos.

- improve checking of valid amount.
  Any number of digits after the decimal point were possible.
  Allow for thousands separators on input.
  Add .00 if no rappen/cents are specified in the input.
  Accoding to the specs, there is a maximum length for amount,
  which was not checked and could be exceeded before.

- Accpet the country entered in the local language.
  Only ISO-2-letter codes and English were accepted, but when
  entering an address, people certainly expect to be able to specify
  the country in the local language, like the rest of the address. Make
  this possible, but only for CH and LI, as the whole libraray is for
  CH/LI QR-bills only.

### CODE REVIEW

**Comment by claudep** [comment]
Wow, many thanks Kurt for these improvement patches.

I generally prefer having independent merge requests, unless the commits are strongly interdependent. So we can review/comment/complete them one by one.
But you don't need to separate those here, I can also review them with this form. Will do during the next days.

Your version numbering is fine, I just didn't give more attention to that as this is really alpha code (mostly an experiment).

File: qrbill/bill.py [hunk 1]
```diff
@@ -123,9 +123,23 @@ def __init__(
         self.account = account
 
         if amount is not None:
+            # remove commonly used thousands separators
+            amount = amount.replace("'","")
+            amount = amount.replace(",","")
```
Comment (claudep) [inline qrbill/bill.py]: I think this is too dangerous to do. Typically, `,` is used as (non-monetary) decimal separator in French, so if someone writes 234,55, it will lead to 23455.00. We can either decide to error out in this case, or guess based on the number of digits after the comma. What do you think?

Moreover, I think we also should accept Decimal values, but that may be in a separate patch.

File: scripts/qrbill [hunk 2]
```diff
@@ -23,7 +23,7 @@ if __name__ == '__main__':
     parser.add_argument('--creditor-city', required=True,
                         help='creditor city')
     parser.add_argument('--creditor-country', default='CH',
-                        help='creditor country')
+                        help='creditor country (in English or 2-letter code)')
```
Comment (claudep) [inline scripts/qrbill]: That doesn't match your code, as `switzerland` is not even accepted. Maybe we can just remove this addition?
I would also like a test for that.

File: qrbill/bill.py [hunk 3]
```diff
@@ -72,6 +72,11 @@ def __init__(self, *, name='', street='', house_num='', pcode=None, city=None, c
         self.city = city
         if not country:
             country = 'CH'
+        # allow users to write the country as if used in an address in the local language
+        if str.lower(country) in ['schweiz', 'suisse', 'svizzera', 'svizra']:
+            country = 'CH'
+        if str.lower(country) in ['fürstentum liechtenstein']:
```
Comment (claudep) [inline qrbill/bill.py]: Please add `liechtenstein` too.

**Comment by claudep** [comment]
I pushed the 4 first commit in master. May I ask you to make two new PRs for the remaining 2 commits?

File: qrbill/bill.py (hunk 1 above)
Comment (kurtkeller) [reply qrbill/bill.py]: Good point, `,` being used as decimal separator in some areas. I'd suggest to accept `'` as thousands separator, but error out on `,`.

I'll create a new pull request.

File: scripts/qrbill (hunk 2 above)
Comment (kurtkeller) [reply scripts/qrbill]: I'm fine with not adding this extension to the help text.

Switzerland in English is accepted already; the iso3166 module accepts all variants it outputs also as input: alpha2 / alpha3 / numeric / name:

```
>>> from iso3166 import countries
>>> countries.get("CH")
Country(name='Switzerland', alpha2='CH', alpha3='CHE', numeric='756', apolitical_name='Switzerland')
>>> countries.get("CHE")
Country(name='Switzerland', alpha2='CH', alpha3='CHE', numeric='756', apolitical_name='Switzerland')
>>> countries.get("756")
Country(name='Switzerland', alpha2='CH', alpha3='CHE', numeric='756', apolitical_name='Switzerland')
>>> countries.get("switzerland")
Country(name='Switzerland', alpha2='CH', alpha3='CHE', numeric='756', apolitical_name='Switzerland')
```

Comment (claudep) [reply scripts/qrbill]: Ah, thanks for reminding me. So if you are able to make a separate PR including tests, this will be fine!

File: qrbill/bill.py (hunk 3 above)
Comment (claudep) [reply qrbill/bill.py]: I guess the same response as above explain that here too.

**Comment by claudep** [comment]
I'm closing this one, eagerly waiting for two other PRs :-)
